ibis/backends/clickhouse/__init__.py

```diff
@@ -423,7 +423,25 @@ def insert(
         elif not isinstance(obj, ir.Table):
             obj = ibis.memtable(obj)
 
-        query = sge.insert(self.compile(obj), into=name, dialect=self.name)
+        # Compare the columns between the target table and the object to be inserted
+        # If they don't match, assume auto-generated column names and use positional
+        # ordering.
+        columns = (
+            obj.columns
+            if not set(parent_cols := self.get_schema(name).names).difference(
+                obj.columns
+            )
+            else parent_cols
```

Comment:
I'll rename these to `target_*` since `parent` is a bit confusing in this context: there's generally tree relationship between the input table and and the target table.

Reply:
Done.